controllers.js

```diff
@@ -0,0 +1,33 @@
+const endpoints = require("./endpoints.json");
+const { fetchTopics, fetchArticleById } = require("./models");
+
+function getTopics(req, res, next) {
+  fetchTopics()
+    .then((topics) => {
+      res.status(200).send({ topics });
+    })
+    .catch((err) => {
+      next(err);
+    });
+}
+
+function getApi(req, res) {
+  res.status(200).send({ endpoints });
+}
+
+function getArticleById(req, res, next) {
+  const { article_id } = req.params;
+  fetchArticleById(article_id)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch((err) => {
+      if (err.status) {
+        res.status(err.status).send({ msg: err.msg });
+      } else {
```

Comment:
It would be good to use `next` in the catch block to send the error back to your _error handlers_ inside `app.js`. You could then send the `res.status(err.status).send({ msg: err.msg })` in your error handler rather than from your catch 
( this would save you from having to repeat this line of code in many controllers )